js/main.js

```diff
@@ -79,62 +79,308 @@ const pictures = generatePhotos();
   let img = document.createElement('img');
   img.classList.add('social__picture');
-  img.src = pictures[0].comments[count].avatar;
-  img.alt = pictures[0].comments[count].name;
-  img.style.width = 35 + 'px';
-  img.style.height = 35 + 'px';
+  img.src = obj.comments[count].avatar;
+  img.alt = obj.comments[count].name;
+  img.style.width = '35px';
+  img.style.height = '35px';
   return img;
 };
 
-const newCommentText = function (count) {
+const createCommentText = function (count, obj) {
   let p = document.createElement('p');
   p.classList.add('social__text');
-  p.textContent = pictures[0].comments[count].message;
+  p.textContent = obj.comments[count].message;
   return p;
 };
 
-const newCommentElements = function () {
-  for (let i = 0; i < pictures[0].comments.length; i++) {
-    let newComment = newCommentItem();
-    newComment.append(newCommentImg(i));
-    newComment.append(newCommentText(i));
+const createCommentList = function (obj) {
+  for (let i = 0; i < obj.comments.length; i++) {
+    let newComment = createCommentElement(obj);
+    newComment.append(createCommentImg(i, obj));
+    newComment.append(createCommentText(i, obj));
     socialCommentsList.append(newComment);
   }
   return socialCommentsList;
 };
 
+const createNewPicture = function (obj) {
+  socialCommentsList.innerHTML = '';
+  bigPicture.querySelector('img').src = obj.photo;
+  bigPicture.querySelector('.likes-count').textContent = obj.likes;
+  bigPicture.querySelector('.comments-count').textContent = obj.comments.length;
+  bigPicture.querySelector('.social__caption').textContent = obj.description;
+  createCommentList(obj);
+};
+
 // заполняю комментари из моки №1
-newCommentElements();
+createNewPicture(pictures[0]);
 
 const socialCommentCount = bigPicture.querySelector('.social__comment-count');
 socialCommentCount.classList.add('hidden');
 const commentsLoader = bigPicture.querySelector('.comments-loader');
 commentsLoader.classList.add('hidden');
 const body = document.querySelector('body');
-body.classList.add('modal-open');
+// body.classList.add('modal-open');
+
+// module4-task1
+// открытие и закрытие окна для редактирования загружаемой фотографии
+const imgUploadForm = document.querySelector('.img-upload__form');
+const uploadFile = imgUploadForm.querySelector('#upload-file');
+const imgUploadOverlay = imgUploadForm.querySelector('.img-upload__overlay');
+const imgUploadOverlayCancel = imgUploadOverlay.querySelector('.img-upload__cancel');
+
+const onUploadPopupEscPress = function (evt) {
+  if (evt.key === 'Escape') {
+    evt.preventDefault();
+    uploadFile.value = '';
+    imgUploadOverlay.classList.add('hidden');
+    body.classList.remove('modal-open');
+  }
+};
+
+const openUploadPopup = function () {
+  imgUploadOverlay.classList.remove('hidden');
+  body.classList.add('modal-open');
+  document.addEventListener('keydown', onUploadPopupEscPress);
+};
+
+uploadFile.addEventListener('change', function () {
```

Comment:
```suggestion
uploadFileInput.addEventListener('change', function () {
```

иначе можно нечайно предположить в это сам файл, а не поле его выбора